packages/util-crypto/src/scrypt/fromU8a.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright 2017-2020 @polkadot/util-crypto authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Params } from './types';
+
+import { u8aToBn } from '@polkadot/util';
+
+interface Result {
+  params: Params,
+  salt: Uint8Array;
+}
+
+export default function scryptFromU8a (data: Uint8Array): Result {
+  const salt = data.subarray(0, 32);
+  const N = u8aToBn(data.subarray(32 + 0, 32 + 4), { isLe: true }).toNumber();
```

Comment:
I wonder if it would make sense to add some sanity checks here to avoid DoSing by some ridiculously-high numbers.
Probably is not an issue in client-side usage, though, since you can always close the tab.
But if we ever use this code serverside, maybe it would make sense to either explicitly timeout the derivation function or put some reasonable limits (like all defaults + 2 maybe?) on the values here?

Reply:
Yes, this is not great an opens up a hole. For now, I'm going to fix it to the defaults until there is a proper plan of attack - need to really look around on best-practices for this. User inputs are always an issue...

```
// FIXME At this moment we assume these to be fixed params, this is not a great idea since we lose flexibility
  // and updates for greater security. However we need some protection against carefully-crafted params that can
  // eat up CPU since these are user inputs. So we need to get very clever here, but atm we only allow the defaults
  // and if no match, bail out
  assert(N === DEFAULT_PARAMS.N && p === DEFAULT_PARAMS.p && r === DEFAULT_PARAMS.r, 'Invalid injected scrypt params found');
```